packages/core/admin/server/services/user.js

```diff
@@ -180,6 +180,19 @@ const findOne = async (id, populate = ['roles']) => {
   return strapi.entityService.findOne('admin::user', id, { populate });
 };
 
+/**
+ * Find one user by its email
+ * @param {string} id  email
+ * @param {string || string[] || object} populate
+ * @returns
+ */
+const findOneByEmail = async (email, populate = []) => {
```

Comment:
shouldn't this call the findOne function defined above ? Do we need to populate the roles or something like that ?

Reply:
The findOne above uses the `entityService.findOne` which can only find by ID, which isn't what we want here

Reply:
The findOne above uses the `entityService.findOne` method which requires an ID (that we don't have)

Reply:
Should you be using the entityService at least then ?

Reply:
I would then have to use smth like `first(await findMany())`. Since we're using s`trapi.query() `in other methods of the service, I assumed it was cleaner to use it here too, wdyt?

Reply:
I think it's alright for now 👍